examples/inspection/plot_model_inspection_cross_validation.py

```diff
@@ -0,0 +1,222 @@
+# to interpret the model performance.
+#
+# Our baseline model has an R2 of around 0.6 which is better than a dummy
+# regressor. We can therefore use this predictive model as a baseline against
+# more advanced machine learning pipelines.
+#
+# To conclude, cross-validation allows us to answer to two questions: are the
+# results reliable and, if it is, how good is the predictive model.
+#
+# Model inspection
+# ................
+#
+# Once we are happy or at least aware of any model limitations, we can
+# investigate the internals of our model. When we performed the
+# cross-validation, we set the parameter `return_estimator` to `True`.
+#
+# It allows us to get the different predictive models trained and tested within
+# cross-validation.
+cv_estimators = cv_results["estimator"]
+
+# %%
+# While the cross-validation allows us to know if our models are reliable, we
+# have not checked if the model at each cross validation fold is similar.
+#
+# Recall that in our case, our baseline model required tuning of the parameter
+# `alpha`. Therefore, up-to-now, we do not know if the optimal `alpha`
+# parameter for all models are similar. In other words, we are asking the
+# question: what is the variance of the `alpha` parameter value across
+# iterations.
+#
+# Indeed, if the `alpha` parameters vary depending on the input data, it
+# might be challenging to put our model in production because we would not be
+# able to fix this hyperparameter.
+#
+# Let's check the `alpha` parameter variance.
+alpha = [est[-1].alpha_ for est in cv_estimators]
+plt.hist(alpha, bins=30, density=True)
+plt.xlabel("Alpha")
+plt.ylabel("Density")
+_ = plt.title("Distribution of alpha parameter \nduring cross-validation")
+
+# %%
+# We see that the regularization parameter, `alpha`, values are centered and
+# condensed around 40. This is a good sign and means that most of the models
+# tuned within the cross-validation had similar `alpha` values.
+#
+# However, not only hyperparameter such as `alpha` should be studied. The model
+# parameter coming out of the fitting process should analyzed. In our case, we
+# used a linear model. These models are parametrized and rely on two
+# parameters: `coef_` and `intercept_`. Therefore, we should analyze the
+# variance of these parameters as well.
+#
+# For the sake of simplicity, we are going to solely look at the `coef_`.
+import pandas as pd
+
+coefficients = pd.DataFrame([est[-1].coef_ for est in cv_estimators], columns=X.columns)
+coefficients
+
+# %%
+import seaborn as sns
+
+plt.figure(figsize=(9, 7))
+sns.stripplot(data=coefficients, orient="h", color="k", alpha=0.5)
+sns.boxplot(data=coefficients, orient="h")
+plt.axvline(x=0, color=".5")
+plt.title("Coefficient values our model")
+_ = plt.subplots_adjust(left=0.3)
+
+# %%
+# We observe that the coefficients vary minimally, meaning that all the
+# trained models are similar. Each individual model is expected to more or less
+# give the same predictions.
+#
```

Comment:
Maybe we could add a note such as:

> This would not necessarily have been the case if the number of training samples had been smaller, if the model had not been properly regularized or if we had strongly dependent features in the data.